packages/svelte/src/internal/client/render.js

```diff
@@ -1056,6 +1076,8 @@ export function bind_value(dom, get_value, update) {
 
 		dom.value = stringify(value);
 	});
+
+	listen_to_form_reset(dom, update);
```

Comment:
Instead of calling update directly, couldn't the respective binding event handlers be reused?

```diff
- dom.addEventListener('input', () => { ... });
+ const on_input = () => { ... };
+ dom.addEventListener('input', on_input);
  ...
- listen_to_form_reset(dom, update);
+ listen_to_form_reset(dom, on_input);
```

After the reset, the current state of the form elements should be what needs to propagate back, so I think messing with default values should ideally not be necessary.

`listen_to_form_reset` then only needs to execute the passed function (possibly delayed to the next event loop if the actual reset has not happened yet, don't know the timing of the `reset` event).